### PR TITLE
test: added unit tests for the ShareUtils utility class.

### PR DESCRIPTION
**Pull request by ardevd**
Using Powermock is neccessary here in order to mock `TextUtils.isEmpty()`.

### CODE REVIEW

File: app/src/test/java/com/nextcloud/talk/utils/ShareUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        when(context.getResources()).thenReturn(resources);
+        when(resources.getString(R.string.nc_share_text)).thenReturn("Join the conversation at %1$s/index.php/call/%2$s");
+        PowerMockito.when(TextUtils.isEmpty(anyString())).thenReturn(true);
+
+        String expectedResult = "Join the conversation at https://my.nextcloud.com/index.php/call/2aotbrjr";
+        assertEquals("Intent string was not as expected",
+                expectedResult, ShareUtils.getStringForIntent(context, "", userUtils, conversation));
+    }
+
+}
```
Comment (mario) [inline app/src/test/java/com/nextcloud/talk/utils/ShareUtilsTest.java]: New line? :)

File: app/src/test/java/com/nextcloud/talk/utils/ShareUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    private final String token = "2aotbrjr";
+
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void getStringForIntent_assertCorrectStringWithoutPassword() {
```
[the remark this answers is not in the document]
Comment (mario) [reply app/src/test/java/com/nextcloud/talk/utils/ShareUtilsTest.java]: Method names :D

[the remark this answers is not in the document]
Comment (ardevd) [reply app/src/test/java/com/nextcloud/talk/utils/ShareUtilsTest.java]: sure. its still WIP so I'm still working on this ;)

**Comment by ardevd** [comment]
@mario I think you can merge this for now. The `getShareApps` method requires some more work in order to test. Primarily, I need to figure out how it's used. :) You can of course hold off on merging this until I get around to it.

Up to you bud:)

**Comment by mario** [comment]
Thanks. I look forward to your future contributions (and finishing this set of tests).

**Pull request merged**